### ISSUE TITLE
Unexpected RedundantCondition analyzing multiple isset checks on an array

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#347]
Expected: No warning

```php
<?php

function testarray(array $data) : void {
    foreach ($data as $item) {
        if (isset($item['foo']['field2']) && isset($item['gar']['field'])) {  //  Unexpected "RedundantCondition - Found a redundant condition when evaluating $item"
            echo "Found\n";
        }
    }
}
```
